data/spells/scripts/house/edit_door_list.lua

```diff
@@ -1,6 +1,8 @@
 function onCastSpell(creature, variant)
-	local house = creature:getTile():getHouse()
-	local doorId = house and house:getDoorIdByPosition(variant:getPosition())
+	local creaturePos = creature:getPosition()
+	local house = Tile(creaturePos):getHouse()
```

Comment:
as far as I remember you can also use aleta grav while you are not in your house (one sqm before your house door), so house will return nil in this case and spell wont work.
My solution:
```
function onCastSpell(creature, variant)
	local creaturePos = creature:getPosition()
	creaturePos:getNextPosition(creature:getDirection())

	local tile = Tile(creaturePos)
	if not tile then
		creature:sendCancelMessage(RETURNVALUE_NOTPOSSIBLE)
		creature:getPosition():sendMagicEffect(CONST_ME_POFF)
		return false
	end

	local house = tile:getHouse()
	local doorId = house and house:getDoorIdByPosition(creaturePos)
	if not doorId or not house:canEditAccessList(doorId, creature) then
		creature:sendCancelMessage(RETURNVALUE_NOTPOSSIBLE)
		creature:getPosition():sendMagicEffect(CONST_ME_POFF)
		return false
	end

	creature:setEditHouse(house, doorId)
	creature:sendHouseWindow(house, doorId)
	return true
end
```